exec/java-exec/src/main/java/org/apache/drill/exec/exception/VersionMismatchException.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.exception;
+
+
+import org.apache.drill.common.exceptions.DrillRuntimeException;
+
+public class VersionMismatchException extends DrillRuntimeException {
+
+  public VersionMismatchException() {
+    super();
+  }
+
+  public VersionMismatchException(Throwable cause) {
+    super(cause);
+  }
+
+  public VersionMismatchException(String message) {
+    super(message);
+  }
+
+  public VersionMismatchException(String message, Throwable cause) {
+    super(message, cause);
+  }
+
+  public VersionMismatchException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
+    super(message, cause, enableSuppression, writableStackTrace);
+  }
+
```

Comment:
No need to create ctors that are not actually used.

Also, because this is very specific, should the exception hold the expected & actual version numbers? Or, take them as args to the ctor and do message formatting?

public VersionMismatchException( String msg, VersionSth expected, VersionSth actual, Exception e ) {
super.( msg + " - Expected: " + expected + ", Actual: " + actual, e );
}